src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php

```diff
@@ -5,19 +5,52 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Intl\Intl;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\Translator;
 
 class LocaleType extends AbstractType
 {
+    /** @var Translator */
+    protected $translator;
+
+    /** @var float */
+    protected $minTranslationProgress;
+
+    /**
+     * @param Translator $translator
+     * @param float      $minTranslationProgress
+     */
+    public function __construct(Translator $translator, $minTranslationProgress)
+    {
+        $this->translator             = $translator;
+        $this->minTranslationProgress = (float) $minTranslationProgress;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults(
-            array(
-                'choices' => Intl::getLocaleBundle()->getLocaleNames('en'),
-            )
-        );
+        $availableLocales = Intl::getLocaleBundle()->getLocaleNames('en');
+        $locales = [];
+
+        if (null == $this->translator) {
```

Comment:
in which case the service `translator` can be null ?
And if it's true, please make a strict condition with `===`